src/lib/cache/abstract_cache.hpp

```diff
@@ -71,7 +85,7 @@ class AbstractCacheImpl {
   // Remove an element from the cache according to the cache algorithm's strategy
   virtual void _evict() = 0;
 
-  size_t _capacity;
+  std::atomic_size_t _capacity;
```

Comment:
Now that the AbstractCache class has "unsafe" methods and atomic members, we should clearly define its concurrency guarantees.